packages/api/tests/lib/test-granule-remove-from-cmr.js

```diff
@@ -322,17 +348,82 @@ test.serial('unpublishGranule() does not update granule if Dynamo write fails',
   await t.throwsAsync(
     unpublishGranule(
       t.context.knex,
-      granule,
+      originalDynamoGranule,
       t.context.granulePgModel,
       fakeGranuleDynamoModel
     )
   );
 
+  t.false(cmrDeleteStub.called);
+  t.like(
+    await t.context.granulesModel.get({ granuleId: originalDynamoGranule.granuleId }),
+    {
+      published: true,
+      cmrLink: originalDynamoGranule.cmrLink,
+    }
+  );
+  t.like(
+    await t.context.granulePgModel.get(t.context.knex, {
+      cumulus_id: pgGranuleCumulusId,
+    }),
+    {
+      published: true,
+      cmr_link: originalPgGranule.cmr_link,
+    }
+  );
+});
+
+test.serial('unpublishGranule() does not update granule if CMR removal fails', async (t) => {
```

Comment:
Nit: It does actually update the Dynamo granule, right? You're just catching the error that CMR might throw and updating the Dynamo record _again_ with the correct `published` status. Maybe we want to say something like `unpublishGranule() maintains the granule status if CMR removal fails`?

Reply:
Hmm I think maybe it's semantics, but yeah what happens is:

- Dynamo record is updated to remove CMR link and set `published: false`
- CMR removal fails
- Error is caught and Dynamo record is reverted to restore CMR link and set `published: true`

So when the test title says `does not update granule`, I mean that the final state of the granule after a CMR removal failure is to unchanged from its initial state. 

Would this change be better?

```suggestion
test.serial('unpublishGranule() does not mark granule as unpublished if CMR removal fails', async (t) => {
```

Reply:
What do you think of https://github.com/nasa/cumulus/pull/2358/commits/6fd776361271648ed8b12234a358bd3f06acf4ac?

Reply:
Yup, that's great and it is 100% semantics 😄